selection_tools/prepare_histograms.C

```diff
@@ -116,44 +144,48 @@ TH1D* get_weights_histogram(TH1D* baseline_hist, double destination_lifetime)
 
 int main(int argc, char *argv[])
 {
-  if(argc != 3){
-    cout<<"Usage: ./prepare_histograms input_path output_path"<<endl;
+  if(argc != 4){
+    cout<<"Usage: ./prepare_histograms input_path output_path ctau_option"<<endl;
     exit(0);
   }
   
   string input_path = argv[1];
   string output_path = argv[2];
+  double ctau_option = atof(argv[3]);
   
   auto input_tree = get_input_tree(input_path);
   
 // load events
   auto event_reader = EventReader(max_events, n_daughters);
   auto events = event_reader.read_events(input_tree);
+
+  bool reduce_hists = false;
+  if(ctau_option != 0.0) reduce_hists = true;
   
-  TH1D *baseline_hist = fill_and_save_histograms(events, output_path);
-  
+  TH1D *baseline_hist = fill_and_save_histograms(events, output_path, reduce_hists);
+
 //  vector<double> destination_lifetimes = {1e-6, 1e-5, 1e-4, 1e-3, 1e-2, 1e-1, 1e0, 1e1, 1e2, 1e3, 1e4}; // mm
-  vector<double> destination_lifetimes = {1e-3, 1e0}; // mm
+  // vector<double> destination_lifetimes = {1e-3, 1e0}; // mm
   
-  for(double destination_lifetime : destination_lifetimes){
+  // for(double destination_lifetime : destination_lifetimes){
     
-    TH1D *weights_hist = get_weights_histogram(baseline_hist, destination_lifetime);
+  //   TH1D *weights_hist = get_weights_histogram(baseline_hist, destination_lifetime);
     
-    // set new event weights
-    for(auto event : events){
-      double event_proper_lifetime = event->get_proper_lifetime();
-      int weight_bin = weights_hist->GetXaxis()->FindFixBin(event_proper_lifetime);
-      double weight = weights_hist->GetBinContent(weight_bin);
-      if(weight < 0){
-        cout<<"Found event for which weight cannot be calculated..."<<endl;
-        weight = 1;
-      }
-      event->weight = weight;
-    }
+  //   // set new event weights
+  //   for(auto event : events){
+  //     double event_proper_lifetime = event->get_proper_lifetime();
+  //     int weight_bin = weights_hist->GetXaxis()->FindFixBin(event_proper_lifetime);
+  //     double weight = weights_hist->GetBinContent(weight_bin);
+  //     if(weight < 0){
+  //       cout<<"Found event for which weight cannot be calculated..."<<endl;
+  //       weight = 1;
+  //     }
+  //     event->weight = weight;
+  //   }
     
-    string lifetime_output_path = replace_all(output_path, ".root", "_ctau-"+to_nice_string(destination_lifetime, 6)+"mm.root");
-    fill_and_save_histograms(events, lifetime_output_path);
-  }
+  //   string lifetime_output_path = replace_all(output_path, ".root", "_ctau-"+to_nice_string(destination_lifetime, 6)+"mm.root");
+  //   fill_and_save_histograms(events, lifetime_output_path);
+  // }
```

Comment:
Similarly here, if you're not sure what to do with this code, just add a flag to turn off its execution, but try not to comment code out.